src/ResourceManager/IotCentral/Commands.IotCentral/IotCentral/NewAzureRmIotCentralApp.cs

```diff
@@ -0,0 +1,137 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.IotCentral.Common;
+using Microsoft.Azure.Commands.IotCentral.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.ResourceManager.Common.Tags;
+using Microsoft.Azure.Management.Internal.Resources;
+using Microsoft.Azure.Management.IotCentral;
+using Microsoft.Azure.Management.IotCentral.Models;
+using System.Collections;
+using System.Collections.Generic;
+using System.Management.Automation;
+using ResourceProperties = Microsoft.Azure.Commands.Management.IotCentral.Properties;
+
+namespace Microsoft.Azure.Commands.Management.IotCentral
+{
+    [Cmdlet(VerbsCommon.New, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "IotCentralApp", SupportsShouldProcess = true)]
+    [OutputType(typeof(PSIotCentralApp))]
+    public class NewAzureRmIotCentralApp : IotCentralBaseCmdlet
+    {
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 2,
+            HelpMessage = "Subdomain for the IoT Central URL. Each application must have a unique subdomain.",
+            ParameterSetName = InteractiveIotCentralParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string Subdomain { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Custom display name for the IoT Central application. Default is resource name.")]
+        [ValidateNotNullOrEmpty]
+        public string DisplayName { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "IoT Central application template name. Default is a custom application.")]
+        [ValidateNotNullOrEmpty]
+        public string Template { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Pricing tier for IoT Central applications. Default value is S1.")]
+        [ValidateNotNullOrEmpty]
+        public PSIotCentralAppSku? Sku { get; set; }
```

Comment:
Why is this type nullable?

Reply:
Also, any reason why this shouldn't be a string with a TabCompleter for the current values?  This would allow SKU values to expand without requiring immediate cmdlet update

Reply:
Done